src/Catalog/Helpers/Utils.cs

```diff
@@ -32,7 +34,15 @@ public static Stream GetResourceStream(string resName)
 
         public static string GetResource(string resName)
         {
-            return new StreamReader(GetResourceStream(resName)).ReadToEnd();
+            if (!_resourceStringCache.ContainsKey(resName))
+            {
+                using (var reader = new StreamReader(GetResourceStream(resName)))
+                {
+                    _resourceStringCache[resName] = reader.ReadToEnd();
```

Comment:
Does this `_resourceStringCache` ever get cleaned? I get the feeling that it could eventually inflate to massive sizes and cause an issue unless we use some sort of LRU logic in here.